src/tests/integration/test_run.py

```diff
@@ -0,0 +1,173 @@
+import uuid
+from functools import partial
+
+import pytest
+from sqlalchemy import create_engine
+import sqlalchemy
+
+from sql.connection import SQLAlchemyConnection, DBAPIConnection
+from sql.run.run import run_statements
+from sql import _testing
+
+
+@pytest.fixture
+def psycopg2_factory():
+    import psycopg2
+
+    config = _testing.DatabaseConfigHelper.get_database_config("postgreSQL")
+
+    return partial(
+        psycopg2.connect,
+        database=config["database"],
+        user=config["username"],
+        password=config["password"],
+        host=config["host"],
+        port=config["port"],
+    )
+
+
+class ConfigAutocommit:
+    autopandas = None
+    autopolars = None
+    autocommit = True
+    feedback = True
+    polars_dataframe_kwargs = {}
+    style = "DEFAULT"
+    autolimit = 0
+    displaylimit = 10
+
+
+class ConfigNoAutocommit(ConfigAutocommit):
+    autocommit = False
+
+
+# TODO: refactor the fixtures so each test can use its own database
+# and we don't have to worry about unique table names
+def gen_name():
+    return f"table_{str(uuid.uuid4())[:8]}"
+
+
+@pytest.mark.xfail(reason="this is failing with sqlalchemy 1.x")
+def test_duckdb_sqlalchemy_doesnt_commit_by_default(tmp_empty):
+    """
+    This test checks that duckdb doesn't commit by default so we're sure that the
+    commit behavior comes from our code
+    """
+    url = "duckdb:///my.db"
+
+    conn_one = create_engine(url).connect()
+    conn_two = create_engine(url).connect()
+
+    name = gen_name()
+    conn_one.execute(sqlalchemy.text(f"CREATE TABLE {name} (id int)"))
+
+    with pytest.raises(sqlalchemy.exc.ProgrammingError) as excinfo:
+        conn_two.execute(sqlalchemy.text(f"SELECT * FROM {name}"))
+
+    assert f"Table with name {name} does not exist!" in str(excinfo.value)
+
+
+def test_postgres_dbapi_doesnt_commit_by_default(setup_postgreSQL, psycopg2_factory):
+    """
+    This test checks that postgres doesn't commit by default so we're sure that the
+    commit behavior comes from our code
+    """
+    import psycopg2
+
+    conn_one = psycopg2_factory()
+    conn_two = psycopg2_factory()
+
+    name = gen_name()
+
+    with conn_one.cursor() as c:
+        c.execute(f"CREATE TABLE {name} (id int)")
+
+    with pytest.raises(psycopg2.errors.UndefinedTable):
+        with conn_two.cursor() as c:
+            c.execute(f"SELECT * FROM {name}")
+
+
+# TODO: duckdb-engine does not support isolation level so we need to test with
+# a database that does (but first verify that this is the case)
+
+
+@pytest.mark.xfail(reason="this is failing with sqlalchemy 1.x")
```

Comment:
can we skip only if sqlalchemy<2 ?

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/755/commits/a4a8569f22b9d8139e6337192288c644b90f9587)